docs/Testing.rst

```diff
@@ -33,6 +33,17 @@ The LLVM lit-based testsuite
 
 * Buildbots run all tests, on all supported platforms.
 
+Testsuite subsets
+-----------------
+
+The testsuite is split into three subsets:
+
+* Primary testsuite, located under ``swift/test``.
+
+* Validation testsuite, located under ``swift/validation-test``.
+
+* Long tests, which are marked with ``REQUIRES: long_test``.
```

Comment:
@modocache IMHO it is a bit subjective. You have to look at the length of time vs the signal. For these tests over time, we have realized that the signal is not worth the cost to run more than once a day.

One additional thing that we have spoken about is defining a specific time budget for different levels of tests. To add something to that testing level, one either has to get the project to expand the time budget or remove a different test.